phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2957,149 +2958,180 @@ public boolean isViewReferenced() {
                 splits = SchemaUtil.processSplits(splits, pkColumns, saltBucketNum, connection.getQueryServices().getProps().getBoolean(
                         QueryServices.FORCE_ROW_KEY_ORDER_ATTRIB, QueryServicesOptions.DEFAULT_FORCE_ROW_KEY_ORDER));
             }
-            MetaDataMutationResult result = connection.getQueryServices().createTable(
-                    tableMetaData,
-                    viewType == ViewType.MAPPED || allocateIndexId ? physicalNames.get(0).getBytes() : null,
-                    tableType, tableProps, familyPropList, splits, isNamespaceMapped, allocateIndexId,
-                    UpgradeUtil.isNoUpgradeSet(connection.getClientInfo()), parent);
+
+			// Modularized this code for unit testing
+            MetaDataMutationResult result = getCreateTableMutationResult(tableMetaData, viewType,
```

Comment:
Why do we need a separate public method for this? Are there any other places we envision it being used?

Reply:
I was stubbing this function earlier in the unit test MetaDataClientTest class. I've changed the test now I'll remove the separate public method.